WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -273,6 +274,33 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 			TransactionStore.AddOrUpdate(tx);
 		}
 
+		if (tx.Confirmed)
+		{
+			// Update for TurboSync - mark internal keys as spent if there is a spender tx and no more coins left on the key.
+			foreach (var spenderKey in myInputs.Select(x => x.HdPubKey).Where(x => x.IsInternal).Distinct())
+			{
+				if (spenderKey.Coins.Any(x => !x.IsSpent()))
+				{
+					// The key still has unspent coins.
+					continue;
+				}
+
+				// All the coins on this key were spent. Mark it as spent and store the block height.
+				if (spenderKey.LatestSpentHeight is null)
```

Comment:
Avoid so much nesting. all these lines look like a:

```c#
spenderKey.LatestSpentHeight = spenderKey.LatestSpentHeight is null || spenderKey.LatestSpentHeight < tx.Height ? tx.Height : 0;
```
Or something like that. You can also use `else if` instead of `else\nif`